### DISCUSSION TITLE
API commands (limit types) to set limit

### DISCUSSION THREAD

**Discussion by Andre7tyone** [tbnobody/OpenDTU#742] [Q&A]
Hello,
I have searched for the limit types for the API command to set the limit.
I have found these limit types (see below), are they correct? I don't want to damage the memory of the inverter
by writing to often a persistant limit, therfore it is important to know the code for the non persitant limits.

Thank you!!

limit_type = 0          AbsoluteNonPersistent
limit_type = 1          RelativeNonPersistent
limit_type = 256      AbsolutePersistent
limit_type = 257      RelativePersistent

**Comment by h3tz** [comment]
just do to set the limit: 

r = requests.post(
                    url = f'http://{dtu_ip}/api/limit/config',
                    data = f'data={{"serial":"{serial}", "limit_type":0, "limit_value":{setpoint}}}',
                    auth = HTTPBasicAuth(dtu_nutzer, dtu_passwort),
                    headers = {'Content-Type': 'application/x-www-form-urlencoded'}
                )

**Comment by h3tz** [comment]
<img width="470" alt="image" src="https://github.com/tbnobody/OpenDTU/assets/6421402/7a6e0367-d79a-413e-8f54-b3aed90d597f">

**Comment by flash69** [comment]
> Hello, I have searched for the limit types for the API command to set the limit. I have found these limit types (see below), are they correct? I don't want to damage the memory of the inverter by writing to often a persistant limit, therfore it is important to know the code for the non persitant limits.
> 
> Thank you!!
> 
> limit_type = 0 AbsoluteNonPersistent limit_type = 1 RelativeNonPersistent limit_type = 256 AbsolutePersistent limit_type = 257 RelativePersistent

Searched the same and found this:

https://github.com/tbnobody/OpenDTU/blob/1f1227fa104cda435df7132ddfa011524a262554/lib/Hoymiles/src/commands/ActivePowerControlCommand.h
![image](https://github.com/tbnobody/OpenDTU/assets/20746829/5d438aca-9eb7-4205-8700-a4f295e0d6eb)

So, should be correct.
